src/collection.rs

```diff
@@ -429,52 +475,263 @@ impl<'a, C: ClientExt> Collection<'a, C> {
+    ///
+    /// If silent is not set to true, the body of the response contains a JSON
+    /// object with the following attributes:
+    ///
+    /// _id contains the document identifier of the newly created document
+    /// _key contains the document key
+    /// _rev contains the document revision
+    /// If the collection parameter waitForSync is false, then the call returns
+    /// as soon as the document has been accepted. It will not wait until
+    /// the documents have been synced to disk.
     ///
-    /// # Note
-    /// this function would make a request to arango server.
+    /// Optionally, the query parameter waitForSync can be used to force
+    /// synchronization of the document creation operation to disk even in
+    /// case that the waitForSync flag had been disabled for the entire
+    /// collection. Thus, the waitForSync query parameter can be used to
+    /// force synchronization of just this specific operations. To use this,
+    /// set the waitForSync parameter to true. If the waitForSync parameter is
+    /// not specified or set to false, then the collection’s default
+    /// waitForSync behavior is applied. The waitForSync query parameter
+    /// cannot be used to disable synchronization for collections that have a
+    /// default waitForSync value of true.
+    ///
+    /// If the query parameter returnNew is true, then, for each generated
+    /// document, the complete new document is returned under the new attribute
+    /// in the result.
     #[maybe_async]
-    pub async fn create_document<T>(&self, _doc: Document<T>) {
-        unimplemented!()
+    pub async fn create_document<T>(
+        &self,
+        doc: T,
+        insert_options: Option<DocumentInsertOptions>,
+    ) -> Result<DocumentResponse<T>, ClientError>
+    where
+        T: Serialize + DeserializeOwned,
+    {
+        let mut url = self.document_base_url.join("").unwrap();
+        let body = serde_json::to_string(&doc)?;
+        let query = qs::to_string(&insert_options).unwrap();
+        url.set_query(Some(query.as_str()));
+        let resp: DocumentResponse<T> =
+            deserialize_response(self.session.post(url, body.as_str()).await?.body())?;
+        Ok(resp)
+    }
+
+    /// Reads a single document
+    /// Returns the document identified by document-id. The returned document
+    /// contains three special attributes: _id containing the document
+    /// identifier, _key containing key which uniquely identifies a document in
+    /// a given collection and _rev containing the revision.
+    #[maybe_async]
+    pub async fn read_document<T>(&self, _key: &str) -> Result<Document<T>, ClientError>
+    where
+        T: Serialize + DeserializeOwned,
+    {
+        self.read_document_with_options(_key, None).await
     }
 
+    #[maybe_async]
+    pub async fn read_document_with_options<T>(
+        &self,
+        _key: &str,
+        read_options: Option<DocumentReadOptions>,
```

Comment:
With the builder pattern, I think we can remove the `Option` wrapper here.

Intuitively, all fields are optional inside an `DocumentXXXOptions`, while the outter `Option` makes no sense.

But I don't think writing `DocumentReadOptions::builder().build()` to fill a read_options is a good API. We can add a `Default` trait for `DocumentReadOptions` and then `Default::default()` to fill read_options.

```rust
pub async fn read_document_with_options<T>(
    &self,
    _key: &str,
    read_options: DocumentReadOptions)
{
        //...
}

impl Default for DocumentReadOptions{
    fn default()->Self{
        Self::builder().build()
    }
}

fn main(){
      coll.read_document_with_options("valid_key", Default::default());
}
```